src/kvirc/sparser/KviIrcServerParser_numericHandlers.cpp

```diff
@@ -3039,6 +3039,30 @@ void KviIrcServerParser::parseNumericSaslFail(KviIrcMessage * msg)
 		pOut->output(KVI_OUT_SERVERINFO, __tr2qs("SASL authentication error: %Q"), &szParam);
 	}
 
+	// Handle fallback if possible for SASL auth failure or dump if user mandates SASL
+	// and no fallback is available
+	if(msg->numeric() == 904)
+	{
+		if(msg->connection()->stateData()->sentSaslMethod() == QStringLiteral("EXTERNAL"))
+		{
+			if(!msg->connection()->target()->server()->saslNick().isEmpty() && !msg->connection()->target()->server()->saslPass().isEmpty())
+			{
+				KviWindow * pOut = static_cast<KviWindow *>(msg->console());
```

Comment:
It doesn't work without static_cast?

Reply:
I literally just copied that from the code that was already there to keep it consistent.